homeassistant/components/recorder/models.py

```diff
@@ -72,7 +72,9 @@ class States(Base):   # type: ignore
     __table_args__ = (Index('states__state_changes',
                             'last_changed', 'last_updated', 'entity_id'),
                       Index('states__significant_changes',
-                            'domain', 'last_updated', 'entity_id'), )
+                            'domain', 'last_updated', 'entity_id'),
+                      Index('states__last_updated', 'last_updated'),
```

Comment:
For single column indices, we can make use of the index=True helper like here: https://github.com/home-assistant/home-assistant/pull/6682/files#diff-d7d7ae12256c8ed7fda7d5847c80f7e3R32